### PR TITLE
Feature/better code

### PR DESCRIPTION
**Pull request by whitx**
# Description

Redesign the code of the windows agent, to be more readable and functional
# Impacts

windows agent
(tested with PEAP/EAP-TLS, Open mode)
# Delete branch after merge

YES 
## Enhancements
- Make code clearer and stop fetching 10 times the profile

### CODE REVIEW

**Comment by whitx** [comment]
@fdurand If you can review the code, finishing with my testing meanwhile

File: parse.py [hunk 1]
```diff
@@ -12,7 +12,7 @@
 from imageBG import bgimg
 import wx
 
-def parsing():
+class models():
```
Comment (julsemaan) [inline parse.py]: Class names are camel cased :
ex : 
SandwichDeDinde
RebootFactory
ZammitFactory (same thing as above)
Mustard

File: parse.py [hunk 2]
```diff
@@ -169,185 +169,211 @@ def parsing():
...
+            add_ca = " -addstore -user \"Root\" "
+            Popen(certutil+add_ca+configure_eap()['ca_file_binary'], shell=True)
+        except:
+            msgbox("The Certificate of Authority could not be installed on your machine, please contact your local support.", "Error")
+            exit(0)	
+    
+class MainPanel(wx.Panel):
+
+	def ExecuteOperations(self, e):
+		profile_xml().fetch_profile()
```
Comment (julsemaan) [inline parse.py]: Why do you fetch the profile to do absolutely nothing with it

File: parse.py [hunk 3]
```diff
@@ -169,185 +169,211 @@ def parsing():
...
+		local_computer().set_env()
+		profile_xml().parse_profile()
```
Comment (julsemaan) [inline parse.py]: Again, you are doing nothing with the return

**Comment by julsemaan** [comment]
Looks like you fetch the profile 10 times per request....
You need to make it object oriented, not just wrap procedural programming in 3 classes

**Pull request closed without merging**